### ISSUE TITLE
[Bug]: @storybook/addon-storyshots TypeError: Cannot read properties of undefined (reading 'location')

### ISSUE THREAD

**Issue by stevepond** [storybookjs/storybook#21311]
### Describe the bug

Following documents for storybook@next v7*, in a NextJS13 Project.  
Getting the following when running the initial test to capture screenshots from stories.

```
yarn test storybook.test.js
```
```` 
TypeError: Cannot read properties of undefined (reading 'location')

      at getSelectionSpecifierFromPath (node_modules/@storybook/preview-api/dist/index.js:67:1115)
      at new UrlStore (node_modules/@storybook/preview-api/dist/index.js:67:1627)
      at new PreviewWeb (node_modules/@storybook/preview-api/dist/index.js:70:1158)
      at Object.start (node_modules/@storybook/preview-api/dist/index.js:83:2133)
      at start (node_modules/@storybook/addon-storyshots/dist/frameworks/react/loader.js:18:45)
      at Object.<anonymous> (node_modules/@storybook/react/dist/index.js:1:3027)
      
 ````
      
 package.json
````
    "dependencies": {
    "@tanstack/react-table": "^8.7.9",
    "@types/node": "18.14.2",
    "@types/react": "18.0.28",
    "@types/react-dom": "18.0.11",
    "eslint": "8.35.0",
    "eslint-config-next": "13.2.1",
    "flowbite": "^1.6.3",
    "flowbite-react": "^0.3.8",
    "next": "13.2.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "^3.2.7",
    "typescript": "4.9.5"
  },
  "devDependencies": {
    "@storybook/addon-essentials": "^7.0.0-beta.56",
    "@storybook/addon-interactions": "^7.0.0-beta.56",
    "@storybook/addon-links": "^7.0.0-beta.56",
    "@storybook/addon-storyshots": "^7.0.0-beta.57",
    "@storybook/blocks": "^7.0.0-beta.57",
    "@storybook/jest": "^0.0.11-next.0",
    "@storybook/nextjs": "^7.0.0-beta.56",
    "@storybook/react": "^7.0.0-beta.56",
    "@storybook/test-runner": "^0.10.0-next.8",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@testing-library/react": "^14.0.0",
    "autoprefixer": "^10.4.13",
    "babel-jest": "^29.4.3",
    "eslint-plugin-storybook": "^0.6.11",
    "postcss": "^8.4.21",
    "postcss-loader": "^7.0.2",
    "storybook": "^7.0.0-beta.56",
    "ts-jest": "^29.0.5"
  }
````
      
 storybook.test.js
````
      // storybook.test.js
      import initStoryshots from "@storybook/addon-storyshots";
      initStoryshots();
````

**Comment by shilman** [comment]
Do you a have a reproduction repo you can share? If not, can you create one? See [how to create a repro](https://storybook.js.org/docs/react/contribute/how-to-reproduce). Thank you! 🙏

Also FYI we're planning to deprecate Storyshots and transition users to the [Test runner](https://github.com/storybookjs/test-runner) or other options. Expect a migration guide soon, but Storyshots will be available until 8.0 at least.

**Comment by vfa-tuanva** [comment]
@stevepond, update 'testEnvironment' in jest.config.js from 'node' to 'jsdom', it try to get Document object from 'jsdom' but your test environment is 'node', so it throw an undefined error
{
  'testEnvironment': 'jsdom'
}

But as @shilman mentioned above, I suggest that you should use [Test runner](https://github.com/storybookjs/test-runner), it is easier to config and also support [DOM Snapshot](https://www.npmjs.com/package/@storybook/test-runner#dom-snapshot-recipe)

**Comment by manuartero** [comment]
Same issue here, this is my package.json

```json
"devDependencies": {
    "@storybook/addon-actions": "^7.0.26",
    "@storybook/addon-docs": "^7.0.26",
    "@storybook/addon-links": "^7.0.26",
    "@storybook/addon-storyshots": "^7.0.27",
    "@storybook/addon-storyshots-puppeteer": "^7.0.27",
    "@storybook/addons": "^7.0.26",
    "@storybook/react": "^7.0.26",
    "@storybook/react-vite": "^7.0.26",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/react-hooks": "^8.0.1",
    "@testing-library/user-event": "^14.4.3",
    ...
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.6.1",
    "jest-environment-jsdom": "^29.6.1",
    "jest-transform-stub": "^2.0.0",
    "prettier": "2.2.1",
    "puppeteer": "^8.0.0",
    "react-test-renderer": "18.1.0",
    "sass": "^1.32.8",
    "storybook": "^7.0.26",
    "ts-jest": "^29.1.1",
    "typescript": "^4.2.4",
    "vite": "^4.4.2",
    "vite-plugin-svgr": "^3.2.0",
    "vite-tsconfig-paths": "^4.2.0"
  },
```

at `jest.config` we've already added `testEnvironment: 'jsdom'`

This is our current `storyshots.test.js`

```js
import initStoryshots from '@storybook/addon-storyshots';
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer';

/**
 * issue related to React 18
 *  -> need react-test-renderer 18.1.0
 * @see https://github.com/storybookjs/storybook/issues/17985
 * ---
 * @see https://github.com/storybookjs/storybook/issues/21311
 */

initStoryshots({ test: imageSnapshot() });

```

**Comment by manuartero** [comment]
I solved this upgrading `storybook` to `7.1.0` (right now we're using `7.1.0-rc.2`)

```json
    "@storybook/addon-actions": "7.1.0-rc.2",
    "@storybook/addon-docs": "7.1.0-rc.2",
    "@storybook/addon-links": "7.1.0-rc.2",
    "@storybook/addon-storyshots-puppeteer": "7.1.0-rc.2",
    "@storybook/addon-storyshots": "7.1.0-rc.2",
    "@storybook/addons": "7.1.0-rc.2",
    "@storybook/react": "7.1.0-rc.2",
    "@storybook/react-vite": "7.1.0-rc.2",
    ...
    "storybook": "7.1.0-rc.2",
 ```

this is our current `storyshots.test.js`

```js
/* eslint-disable no-undef */

import initStoryshots from '@storybook/addon-storyshots';
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer';

initStoryshots({
  test: imageSnapshot({
    storybookUrl: process.env.STORYBOOK_URL || 'http://localhost:6006',
    getMatchOptions: () => ({ failureThreshold: 0.1 }),
  }),
});
```
